homeassistant/components/fritz/config_flow.py

```diff
@@ -227,9 +247,10 @@ async def async_step_user(
         if user_input is None:
             return self._show_setup_form_init()
         self._host = user_input[CONF_HOST]
-        self._port = user_input[CONF_PORT]
+        self._port = user_input.get(CONF_PORT)
```

Comment:
place here the logic about the default ports from above
```suggestion
        if (port := user_input.get(CONF_PORT)) is None:
            self._port = DEFAULT_HTTPS_PORT if user_input[CONF_SSL] else DEFAULT_HTTP_PORT
        else:
            self._port = port
```